### DISCUSSION TITLE
Medusajs/inventory inventoryLevelService can't be retrieved from @medusajs/inventory

### DISCUSSION THREAD

**Discussion by xmdcode** [medusajs/medusa#5060] [Q&A]
Hello, 

I am trying to develop a plugin and i am in the process of making a subscriber that listens on 'inventory-level.updated'.

This item returns the id of the inventory level updated.

I have search and did my research on what service to use so i can give the inventory-level id and retrieve the inventory-item/variant or product.

I couldn't find any service that suits my needs (because one of them ProductVariantInventoryItem needs variantid + inventory item id).

I search on inventory module and i found inventoryLevelService that you can give the id i get from the subscriber and get back the inventory item id BUT it doesn't get exported from @medusajs/inventory.

So i need to copy paste the service and make it from the start or is there any plan on the future this service being exported so i can use it?

**Comment by srindom** [comment]
You should use the InventoryService exported from @medusajs/inventory to do this. For example, to retrieve the inventory level, you can do:

```javascript
// Get Inventory Item
const [levels, count] = await inventoryService.listInventoryLevels({ id })
const inventoryItemId = levels[0].inventory_item_id

// Get Variant IDs associated with inventory item
const variantInventory = await productVariantInventoryService.listByItem([inventoryItemId])
const associatedVariantIds = variantInventory.map(vi => vi.variant_id)
```

**Reply by alsherif-khalaf** [reply]
How Can I do this in next.js ?

**Comment by xmdcode** [comment]
> You should use the InventoryService exported from @medusajs/inventory to do this. For example, to retrieve the inventory level, you can do:
> 
> ```js
> // Get Inventory Item
> const [levels, count] = await inventoryService.listInventoryLevels({ id })
> const inventoryItemId = levels[0].inventory_item_id
> 
> // Get Variant IDs associated with inventory item
> const variantInventory = await productVariantInventoryService.listByItem([inventoryItemId])
> const associatedVariantIds = variantInventory.map(vi => vi.variant_id)
> ```
Well i tried to do that but it doesn't get exported from @medusajs/inventory.

I found out that indeed the service exists but the types are not right so for that they are not exported for usage.

**Comment by xmdcode** [comment]
I did export the services so i can use them but i get this error now 

AwilixResolutionError: Could not resolve 'inventoryLevelService'.

Resolution path: inventoryLevelService
